### ISSUE TITLE
Supported platforms?

### ISSUE THREAD

**Issue by njbrown09** [nxrighthere/ENet-CSharp#2]
What platforms are supported? Thanks.

**Comment by nxrighthere** [comment]
Windows, Linux, and MacOS are supported. The native library was tested on mobile platforms Android and iOS, but the C# wrapper itself was not. I believe it should work there as well.

**Comment by njbrown09** [comment]
Does the provided release contain a native binary that works on all platforms?

**Comment by nxrighthere** [comment]
Nope. If you want, I can ask a guy who knows how to compile it for the other platforms today.

**Comment by njbrown09** [comment]
Please do

**Comment by nxrighthere** [comment]
To compile the native library for Android, you need to download the [NDK](https://developer.android.com/ndk/downloads/). Files for building added to the [repository](https://github.com/nxrighthere/ENet-CSharp/tree/master/Source/Native/jni).

Run this command `ndk-build NDK_PROJECT_PATH="path_to_enet"`

![NDK building](https://user-images.githubusercontent.com/8279330/44647933-f4de6800-a9f8-11e8-8736-59b459d0ebb0.PNG)

Copy the compiled libraries to the appropriate [plugin folder](https://docs.unity3d.com/Manual/PluginInspector.html).

How to build the native library for iOS is still a question for me.

**Comment by nxrighthere** [comment]
Just discovered [this](https://github.com/hzj629206/unity-build-libenet/blob/master/build-libenet-ios.sh) script. I think that it can be used with minor changes.

**Comment by njbrown09** [comment]
Do you know how to build it on ios?

**Comment by nxrighthere** [comment]
Read the comments above carefully, I already mentioned this.

**Comment by njbrown09** [comment]
ah

**Comment by nxrighthere** [comment]
For iOS you will need [XCode](https://developer.apple.com/xcode/) to [compile](https://docs.unity3d.com/Manual/PluginsForIOS.html) the project with the native library. Support for iOS is added to the wrapper in https://github.com/nxrighthere/ENet-CSharp/commit/a5e3562a6328e251c8cb893d1d0f31cf85f74ef1.

**Comment by wobes1** [comment]
Do we have x32 support?

**Comment by stang-tgs** [comment]
For anyone that ends up here, for iOS, the easiest now is to use CMake 3.14+ which supports iOS out of the box.  A sample command line looks like: 

`cmake . -DCMAKE_SYSTEM_NAME=iOS "-DCMAKE_OSX_ARCHITECTURES=arm64;arm64e" -DCMAKE_OSX_DEPLOYMENT_TARGET=9.3 -DENET_STATIC=1 -DENET_LZ4=1 -GXcode`.

More details on CMake for iOS [can be found here](https://cmake.org/cmake/help/v3.14/manual/cmake-toolchains.7.html#cross-compiling-for-ios-tvos-or-watchos).

I believe potentially, Android can leverage CMake as well, but I didn't bother since @nxrighthere already has a working example shown in his above comment.

**Comment by nxrighthere** [comment]
Thanks for this, very helpful since I'm not using Mac.

**Comment by qiqiqidalao** [comment]
I used libenet.so on the android platform, but after the connection was created, the server did not receive it. Through debugging, the client.Service(0, out netEvent) of the Service method call in ClientNetwork always returns 0.
